### PR TITLE
Lint HTML and markdown with prettier

### PR DESCRIPTION
**Pull request by cthoyt**
This PR runs `npx prettier --write .` to reformat HTML and markdown in a principled, deterministic way. This will make it easier to make updates that have smaller diffs, such as #2042. It adds a note on how to do this in the README-sitedev.md

It also adds a `.prettierignore` file that lists some file types and directories to ignore.

This PR **does not** change how anything looks on the website. However, in 3 cases (see 5ec2d01) it fixes some invalid HTML.

### CODE REVIEW

File: _includes/navbar.html
Comment (matentzn) [inline _includes/navbar.html:159]: Is this how the new linters like it? Looks a bit odd, do you know the motivation for it?

File: README-sitedev.md
Comment (matentzn) [inline README-sitedev.md:153]: Can we get a bit more detail here? Dont we have to install npx prettier first?

Also, how do you see this getting kept up to date? Any chance for a github action that runs this once in a while? Or whenever and md/html file is changed?

**Review by matentzn — APPROVED** [review]
Man this is awesome, I was always wondering about markdown linters, and here is my answer. I checked every single file with an 87% rigour, and it looked great. I have two questions and they are not blockers on merging this, but I will wait for a moment for you to answer before merging.

File: _includes/navbar.html
Comment (cthoyt) [reply _includes/navbar.html:159]: I guess so. We don't have to argue with what the linter does, assuming it's deterministic, even if we don't really like it ourselves.

Comment (matentzn) [reply _includes/navbar.html:159]: No problem, I agree! Just curious.

File: README-sitedev.md
Comment (cthoyt) [reply README-sitedev.md]: I just update the installation docs. There are indeed github actions to automatically run this, but I didn't want to make a first PR too complicated

**Comment by matentzn** [comment]
Awesome!

**Pull request merged**
